iam.tf

```diff
@@ -1,20 +1,17 @@
-locals {
-  irsa_role_create = var.enabled && var.service_account_create && var.irsa_role_create
-}
-
 data "aws_iam_policy_document" "this" {
-  count = (local.irsa_role_create && var.irsa_policy_enabled) || (local.eks_pod_identity_role_create && var.eks_pod_identity_policy_enabled) ? 1 : 0
+  count = (var.enabled && (var.irsa_policy == null || var.pod_identity_policy == null)) ? 1 : 0
 
-  # https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/v2.12.0/docs/install/iam_policy.json
+  # https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/v2.11.0/docs/install/iam_policy.json
```

Comment:
We should retain v2.12.0 version of the policy

Reply:
Policy is intact, only comment was wrong.